src/main/kotlin/org/wycliffeassociates/otter/common/domain/resourcecontainer/FileUsfmExtensions.kt

```diff
@@ -0,0 +1,39 @@
+package org.wycliffeassociates.otter.common.domain.resourcecontainer
+
+import org.wycliffeassociates.otter.common.collections.tree.Tree
+import org.wycliffeassociates.otter.common.collections.tree.TreeNode
+import org.wycliffeassociates.otter.common.data.model.Collection
+import org.wycliffeassociates.otter.common.data.model.Content
+import org.wycliffeassociates.otter.common.domain.usfm.ParseUsfm
+import java.io.File
+import java.io.IOException
+
+fun File.parseUSFMToChapterTrees(projectSlug: String): List<Tree> {
```

Comment:
Perhaps I've unintentionally done something like this before and set a precedent, but I feel like this is a huge abuse of power that extension methods allow.

In some sense, extension methods are almost like hacking away with reflection, though certainly in a less resource offensive way instead inflating the amount of static methods. Extension methods are a nice way to get rid of the "miscellaneous drawer" of a Utils.java class that stores functions like "captialize(string: String) : String" and just adds this functionality to all strings in a set scope, which is objectively a better alternative than ending up having virtually every class in your application tightly coupled to a utils class.

However, in this situation, we're throwing out all good practices and purposes of object oriented design, which encapsulates functionality to a modular, composable construct (and also, more clear code organization), instead preferring a purely procedural programming style, but with sugar to achieve an infix notation (of which, actual infix notation in kotlin is possible to define in a case like this where there is a single parameter).

As previously discussed, the ParseUsfm class likely needs to change, perhaps to be a subclass of some more abstract ParseMarkup class which parses into something that is better defined than the UsfmDocument class.

If this toChapterTrees method was an extension method of UsfmDocument or its eventual successor, I might be inclined to accept its existence as an extension method limtited to the persistence package, though I also might just prefer it to be a method of said class.